onnxruntime/core/providers/cpu/tensor/transpose.cc

```diff
@@ -543,6 +586,14 @@ Status TransposeBase::DoTranspose(const std::vector<size_t>& permutations, const
     status = ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Mismatched data types between input and output Tensors. ",
                              input_type, " != ", output_type);
   } else {
+    TensorShape shape = input_shape_override ? *input_shape_override : input.Shape();
+    if (IsReshape(permutations, shape.GetDims())) {
+      // As long as the dims with values > 1 stay in the same order, it's a reshape.
+      // Example: Shape=(1,1,1024,4096) -> perm=(2,0,3,1).
+      CopyCpuTensor(&input, &output);
```

Comment:
Continuing from my comment above about 1-D tensors - are there any other cases where transposing just boils down to copying the input buffer to the output buffer as is ? For example, cases where perm ordering is in essence the same ordering as the input ? For 1D inputs, perm = [0] and for 2D inputs, perm = [0, 1], etc. I think, while we are at this "resphape" special-casing, we may handle that special case as well ?

Reply:
This is already handled isn't it? Won't IsReshape will return true if the perms are all in the same order?

---
In reply to: [520128687](https://github.com/microsoft/onnxruntime/pull/5550#discussion_r520128687) [](ancestors = 520128687)

Reply:
Actually yeah - sorry - I think I didn't think it through properly